### PR TITLE
fixes #7230 - rails 4 compatibility

### PR DESCRIPTION
**Pull request by unorthodoxgeek**
related to https://github.com/theforeman/foreman/pull/2870

### CODE REVIEW

File: db/seeds.d/50-bootdisk_templates.rb [hunk 1]
```diff
@@ -1,8 +1,8 @@
-kind = TemplateKind.find_or_create_by_name('Bootdisk')
+kind = TemplateKind.find_or_create_by(:name => 'Bootdisk')
 
 ProvisioningTemplate.without_auditing do
   content = File.read(File.join(ForemanBootdisk::Engine.root, 'app', 'views', 'foreman_bootdisk', 'host.erb'))
-  tmpl = ProvisioningTemplate.find_or_create_by_name(
+  tmpl = ProvisioningTemplate.find_or_create_by(
```
Comment (domcleal) [inline db/seeds.d/50-bootdisk_templates.rb]: This isn't equivalent, it's find or creating by all of those attributes and not solely by name.

File: db/seeds.d/50-bootdisk_templates.rb [hunk 2]
```diff
@@ -11,13 +11,13 @@
   tmpl.attributes = {
     :template => content,
     :default  => true,
-    :vendor   => "Foreman boot disk",
-    :locked   => true
+    :vendor   => "Foreman boot disk"
   }
+  tmpl.locked = true
```
Comment (domcleal) [inline db/seeds.d/50-bootdisk_templates.rb]: Why's that moved?

File: db/seeds.d/50-bootdisk_templates.rb [hunk 3]
```diff
@@ -11,13 +11,13 @@
...
   tmpl.save!(:validate => false) if tmpl.changes.present?
 
   content = File.read(File.join(ForemanBootdisk::Engine.root, 'app', 'views', 'foreman_bootdisk', 'generic_host.erb'))
-  tmpl = ProvisioningTemplate.find_or_create_by_name(
+  tmpl = ProvisioningTemplate.find_or_create_by(
```
Comment (domcleal) [inline db/seeds.d/50-bootdisk_templates.rb]: ditto

File: lib/foreman_bootdisk.rb [hunk 4]
```diff
@@ -1,3 +1,4 @@
+require 'foreman_bootdisk/engine'
```
Comment (domcleal) [inline lib/foreman_bootdisk.rb]: This duplicates line 7.

File: db/seeds.d/50-bootdisk_templates.rb (hunk 2 above)
Comment (unorthodoxgeek) [inline db/seeds.d/50-bootdisk_templates.rb]: locked is a protected attribute, can't be mass assigned

File: lib/foreman_bootdisk.rb (hunk 4 above)
Comment (unorthodoxgeek) [inline lib/foreman_bootdisk.rb]: removed line 7, makes no sense to limit to rails 3.

File: db/seeds.d/50-bootdisk_templates.rb [hunk 5]
```diff
@@ -1,33 +1,33 @@
-kind = TemplateKind.find_or_create_by_name('Bootdisk')
+kind = TemplateKind.find_or_create_by(:name => 'Bootdisk')
```
Comment (domcleal) [inline db/seeds.d/50-bootdisk_templates.rb]: .find_or_create_by isn't a valid method on Rails 3, so you'll either be limiting the plugin to Rails 4 or you need to switch to a compatible syntax such as .where(..).first_or_create.

File: db/seeds.d/50-bootdisk_templates.rb [hunk 6]
```diff
@@ -1,33 +1,33 @@
...
   tmpl.save!(:validate => false) if tmpl.changes.present?
 
   content = File.read(File.join(ForemanBootdisk::Engine.root, 'app', 'views', 'foreman_bootdisk', 'generic_host.erb'))
-  tmpl = ProvisioningTemplate.find_or_create_by_name(
-    :name => 'Boot disk iPXE - generic host',
-    :template_kind_id => kind.id,
-    :snippet => false,
-    :template => content
-  )
+  tmpl = ProvisioningTemplate.find_or_create_by(:name => 'Boot disk iPXE - generic host')  do |template|
```
Comment (domcleal) [inline db/seeds.d/50-bootdisk_templates.rb]: double space before `do`

**Comment by domcleal** [comment]
There are some unaddressed comments above, could you update the PR to fix them please?  It also needs rebasing.

**Comment by unorthodoxgeek** [comment]
updated

**Comment by domcleal** [comment]
Obsoleted by 296dcfc9f8d3af6b6c9be8d60cdd92f757ba2b49.

**Pull request closed without merging**